Demo/Demo/PayPalWebPayments/PayPalWebViewModel/PayPalWebViewModel.swift

```diff
@@ -44,51 +44,106 @@ class PayPalWebViewModel: ObservableObject {
+        do {
+            DispatchQueue.main.async {
+                self.state.authorizedOrderResponse = .loading
+            }
+            let payPalClientMetadataID = payPalDataCollector?.collectDeviceData()
+            let order = try await DemoMerchantAPI.sharedService.authorizeOrder(
+                orderID: orderID,
+                selectedMerchantIntegration: selectedMerchantIntegration,
+                payPalClientMetadataID: payPalClientMetadataID
+            )
+            DispatchQueue.main.async {
+                self.state.authorizedOrderResponse = .loaded(order)
+            }
+        } catch {
+            DispatchQueue.main.async {
+                self.state.authorizedOrderResponse = .error(message: error.localizedDescription)
+            }
+            print("❌ Failed to authorize order: \(error.localizedDescription)")
+        }
+    }
 
     func paymentButtonTapped(funding: PayPalWebCheckoutFundingSource) {
         Task {
             do {
-                self.updateState(.loading)
+                DispatchQueue.main.async {
+                    self.state.approveResultResponse = .loading
+                }
                 payPalWebCheckoutClient = try await getPayPalClient()
                 guard let payPalWebCheckoutClient else {
                     print("Error initializing PayPalWebCheckoutClient")
                     return
                 }
 
-                if let orderID {
+                if let orderID = state.createOrder?.id {
                     let payPalRequest = PayPalWebCheckoutRequest(orderID: orderID, fundingSource: funding)
                     payPalWebCheckoutClient.start(request: payPalRequest) { result, error in
                         if let error {
-                            if error == PayPalError.checkoutCanceledError {
-                                print("Canceled")
-                                self.updateState(.idle)
-                            } else {
-                                self.updateState(.error(message: error.localizedDescription))
+                            DispatchQueue.main.async {
+                                if error == PayPalError.checkoutCanceledError {
+                                    print("Canceled")
+                                    self.state.approveResultResponse = .idle
+                                } else {
+                                    self.state.approveResultResponse = .error(message: error.localizedDescription)
+                                }
                             }
                         } else {
-                            self.updateState(.success)
-                            self.checkoutResult = result
+                            DispatchQueue.main.async {
+                                self.state.approveResultResponse = .loaded(PayPalPaymentState.ApprovalResult(id: orderID, status: "COMPLETED"))
```

Comment:
I think this is actually an error that was there before, the status on success url should be "APPROVED" not "COMPLETED"

Reply:
Looks good now! I would just resolve the 7 lint warnings.